### DISCUSSION TITLE
Is possible to define a store independently from a block?

### DISCUSSION THREAD

**Discussion by johnhooks** [WordPress/gutenberg#64643] [Interactivity API]
After reading through the [wp-movies-demo](https://github.com/WordPress/wp-movies-demo/tree/main) example, I have a question about defining Interactivity API stores.

I noticed that some blocks rely on the state properties or actions defined in other blocks. For example the [movie-trailer-button](https://github.com/WordPress/wp-movies-demo/blob/dde4b500dfb02ba85e49ec92a1b6201fd83741f0/src/blocks/interactive/movie-trailer-button/render.php#L35) calls `setVideo` which is defined in  [video-player](https://github.com/WordPress/wp-movies-demo/blob/dde4b500dfb02ba85e49ec92a1b6201fd83741f0/src/blocks/interactive/video-player/view.js#L16-L20).

Is possible to define a store independently from a block, so that can be included as a dependency for multiple blocks? This would keep all the shared logic for a store together, rather than being defined in pieces across multiple blocks. Otherwise a block can have a hidden reliance on state or actions defined in another block, which might not be included on the page.

**Comment by luisherranz** [comment]
Absolutely, blocks and stores are independent, and you can organize them however you want. For example, you could have a single store defined in one file for all your blocks. Or you could have multiple store parts defined in different files that share the same namespace. Or you could define two different stores for a single block. 

It is up to you, depending on what you want to achieve and how your blocks relate to each other. The Interactivity API does not impose any restrictions.

**Reply by johnhooks** [reply]
Awesome, thank you for the confirmation. All the examples showed the store in the view script of a block. After considering how to create a global store that needs to be fully initialized, but shared between multiple blocks it didn't fit any of the examples I had seen.

**Reply by luisherranz** [reply]
I want to prepare a guide for the new [Core Concepts section](https://developer.wordpress.org/block-editor/reference-guides/interactivity-api/core-concepts/) of the Interactivity API where I want to explain how namespaces and scopes work. I'll try to include more examples of this kind.

**Reply by spencerfinnell** [reply]
> you could have a single store defined in one file for all your blocks. 

Noting that I ended up in this scenario myself, so an example of this being an acceptable practice could be helpful.
